source/MaterialXCore/Document.cpp

```diff
@@ -605,6 +605,27 @@ void Document::upgradeVersion()
         minorVersion = 36;
     }
 
+    // Upgrade to versions 1.37 and up
+    if ((majorVersion == 1 && minorVersion < 37) || majorVersion < 1)
```

Comment:
This code is used to convert code prior to 1.37 to 1.37 and above. Namely it converts NodeDefs with types to NodeDefs with a child output called "out" (or in the multi-output case simply removes the type).